cdap-watchdog/src/main/java/co/cask/cdap/logging/gateway/handlers/LogHandler.java

```diff
@@ -103,7 +103,22 @@ private void doGetLogs(HttpResponder responder, LoggingContext loggingContext,
                          long fromTimeSecsParam, long toTimeSecsParam, boolean escape, String filterStr,
                          @Nullable RunRecordMeta runRecord) {
     try {
-      TimeRange timeRange = parseTime(fromTimeSecsParam, toTimeSecsParam, responder);
+      long readFromTimeMillis;
+      long currentTime = System.currentTimeMillis();
+      long readToTimeMillis = currentTime;
+
+      // If runId not specified, fetch past 1 hour logs
+      if (runRecord == null) {
+        readFromTimeMillis = currentTime - TimeUnit.HOURS.toMillis(1);
+      } else {
+        readFromTimeMillis =  TimeUnit.SECONDS.toMillis(runRecord.getStartTs());
+        if (runRecord.getStopTs() != null) {
+          readToTimeMillis = TimeUnit.SECONDS.toMillis(runRecord.getStopTs() + 1);
+        }
+      }
```

Comment:
Yes. But if runRecord is null, `TimeUnit.SECONDS.toMillis(runRecord.getStopTs() + 1)` will fail.